imap/util.c

```diff
@@ -909,15 +909,19 @@ int imap_wait_keepalive (pid_t pid)
 
 void imap_allow_reopen (CONTEXT *ctx)
 {
-  IMAP_DATA* idata = ctx->data;
-  if (ctx && ctx->magic == MUTT_IMAP && idata->ctx == ctx)
+  IMAP_DATA* idata;
+  if (ctx && ctx->magic == MUTT_IMAP && (idata = ctx->data)->ctx == ctx)
```

Comment:
same comment as before — and below, that kind of assignment is better done outside of the `if`. ☺